src/stan/io/deserializer.hpp

```diff
@@ -581,6 +581,64 @@ class deserializer {
     return ret;
   }
 
+  /**
+   * Return the next zero sum vector of the specified size (using one fewer
+   * unconstrained scalars), incrementing the specified reference with the
+   * log absolute Jacobian determinant (no adjustment, in this case).
+   *
+   * <p>See <code>stan::math::sum_to_zero_constrain(Eigen::Matrix,T&)</code>.
+   *
+   * @tparam Ret The type to return.
+   * @tparam Jacobian Whether to increment the log of the absolute Jacobian
+   * determinant of the transform.
+   * @tparam LP Type of log probability.
+   * @tparam Sizes A parameter pack of integral types.
+   * @param lp The reference to the variable holding the log
+   * @param size Number of cells in zero sum vector to generate.
+   * @return The next zero sum of the specified size.
+   * @throws std::invalid_argument if number of dimensions (`k`) is zero
+   */
+  template <typename Ret, bool Jacobian, typename LP,
+            require_not_std_vector_t<Ret>* = nullptr>
+  inline auto read_constrain_sum_to_zero(LP& lp, size_t size) {
+    stan::math::check_positive("read_sum_to_zero", "size", size);
+    return stan::math::sum_to_zero_constrain<Jacobian>(
+        this->read<Ret>(size - 1), lp);
```

Comment:
Should `size - 1` be done at the compiler level? Since we would expect size + 1 to be returned

Reply:
I was just going off of what `simplex` does in the same case (line 547), which does the -1